src/foam/parse/NotExpression.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.parse;
+
+import java.util.List;
+
+import foam.lib.parse.Alt;
+import foam.lib.parse.Literal;
+import foam.lib.parse.PStream;
+import foam.lib.parse.ParserContext;
+import foam.lib.parse.ProxyParser;
+import foam.mlang.MLang;
+import foam.mlang.predicate.Predicate;
+
+public class NotExpression extends ProxyParser {
+
+  public NotExpression() {
+    super(
+          new Alt(           
+            new Literal("NOT"),
+            new Literal("Not"),
+            new Literal("not"),
+            new Literal("-")));
+  }
+
+  @Override
+  public PStream parse(PStream ps, ParserContext x) {
+    ps = super.parse(ps, x);
+    if (ps == null)
+      return null;
+
+    if (ps.value().toString().toLowerCase().equals("not") || ps.value().toString().equals("-")) {

```

Comment:
You shouldn't need to check ps.value(), super.parse() will return null if the value wasn't one of your specified values of NOT/Not/not/-

Also instead of adding all the different capitalizations of "not"  Let's add a LiteralIC class under foam/lib/parse which is similar to Literal() but will ignore the case of the argument.  Then we can simplify this to just new Alt(new LiteralIC("not"), new Literal("-"));